### PR TITLE
pdns: remove duplicated dns2_tolower() function

### PR DESCRIPTION
**Pull request by tfarina**
There is already a version of dns2_tolower() in misc.hh, called
simply dns_tolower(), but due to some inclusion issues it was necessary
to extract it into another header file (named ascii.hh).

### CODE REVIEW

**Comment by Habbie** [comment]
Travis is unhappy with your patch. Can you check the logs and find out why?

**Comment by tfarina** [comment]
I know why it is unhappy, but I don't know autoconf well enough to change the Makefiles to add ascii.hh to the include path so the compiler could find it.

**Comment by Habbie** [comment]
You just need symlinks in the recursordist and dnsdistdist dirs.

**Comment by tfarina** [comment]
Done. Symlinks added.

**Comment by Habbie** [comment]
Travis still unhappy (I did not check why).

**Comment by Habbie** [comment]
Also, if you do requested fixes, please push additional commits, this makes review easier.

**Comment by tfarina** [comment]
dnsname.hh:37:20: fatal error: ascii.hh: No such file or directory

**Comment by rgacogne** [comment]
So, you need to add `ascii.hh` to the `dnsdist_SOURCES` variable in `pdns/dnsdistdist/Makefile.am` and the `pdns_recursor_SOURCES` one in `pdns/recursordist/Makefile.am`, otherwise they are not included in the dnsdist and recursor tarballs.

**Comment by tfarina** [comment]
Done. It passed the trybots. Thanks Remi.

File: pdns/Makefile.am [hunk 1]
```diff
@@ -133,6 +133,7 @@ EXTRA_PROGRAMS = \
 
 pdns_server_SOURCES = \
 	arguments.cc arguments.hh \
+        ascii.hh \
```
Comment (pieterlexis) [inline pdns/Makefile.am]: Please replace with a tab to match style

**Comment by tfarina** [comment]
Done.

**Comment by pieterlexis** [comment]
hi @tfarina,

Pleas ealso do this to the other `Makefile.am`'s from this PR. They have the same issue 😉

**Comment by tfarina** [comment]
Done.

**Pull request merged**